### PR TITLE
core.cpuid: emit fewer fixups

### PR DESCRIPTION
**Pull request by WalterBright**
The idea is to:

1. consolidate private globals into a struct, reducing the number of pointless symbols in the object file
2. use a pointer to the struct instance to access the globals, which reduces the fixups in the executable image, making for faster loading and a smaller executable file size.

Worthwhile because core.cpuid is compiled into every D executable.

### CODE REVIEW

**Comment by WalterBright** [comment]
Needs careful review, because this code is pretty much untestable.

File: src/core/cpuid.d [hunk 1]
```diff
@@ -980,57 +1002,57 @@ shared static this()
...
-    _sse2 =           (features&SSE2_BIT)!=0;
-    _sse3 =           (miscfeatures&SSE3_BIT)!=0;
-    _ssse3 =          (miscfeatures&SSSE3_BIT)!=0;
-    _sse41 =          (miscfeatures&SSE41_BIT)!=0;
-    _sse42 =          (miscfeatures&SSE42_BIT)!=0;
-    _sse4a =          (amdmiscfeatures&SSE4A_BIT)!=0;
-    _aes =            (miscfeatures&AES_BIT)!=0;
-    _hasPclmulqdq =   (miscfeatures&PCLMULQDQ_BIT)!=0;
-    _hasRdrand =      (miscfeatures&RDRAND_BIT)!=0;
+    _vendor =         cast(string)cf.vendorID;
```
Comment (DmitryOlshansky) [inline src/core/cpuid.d]: would it make sense to put immutables into a struct too? Then these lines are trivially re-written as `icf = Immortals(cast(string)cf.vendorID, ...); `

Comment (WalterBright) [inline src/core/cpuid.d]: Yes, but I wished to limit the scope of this PR to make it easier to review.

Comment (DmitryOlshansky) [inline src/core/cpuid.d]: Separate commit will make it just as easy. Well doing the next PR will work too

**Comment by WalterBright** [comment]
rebased

**Comment by WalterBright** [comment]
Any reason why this is not being pulled?

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by WalterBright** [comment]
thanks! (I don't want to rebase it again!)

**Comment by DmitryOlshansky** [comment]
> thanks! (I don't want to rebase it again!)

No problem, but consider for a moment that a typical pull stays for month or two and survives 3-4 rebases

**Pull request merged**